File: JS/Diferenca.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./Diferenca.js')

TEST('Starting Diferenca test...', (t) => {
    //Test 1
    t.assert(OBJ.showDifference(5, 6, 7, 8) === -26, "Expect: -26")
    //Test 2
    t.assert(OBJ.showDifference(0, 0, 7, 8) === -56, "Expect: -56")
    //Test 3
    t.assert(OBJ.showDifference(5, 6, -7, 8) === 86, "Expect: 86")

    t.end()
})

File: GO/Diferenca_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestDiferenca(t *testing.T) {
	tables := []struct {
		A    int
		B    int
		C    int
		D    int
		want int
	}{
		{5, 6, 7, 8, -26},
		{0, 0, 7, 8, -56},
		{5, 6, -7, 8, 86},
	}

	for _, table := range tables {
		got := showDifference(table.A, table.B, table.C, table.D)

		if got != table.want {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
